src/types/entity_id.ts

```diff
@@ -0,0 +1,25 @@
+const entityIdRegex = /^([a-f]|[0-9]){8}-([a-f]|[0-9]){4}-([a-f]|[0-9]){4}-([a-f]|[0-9]){4}-([a-f]|[0-9]){12}$/;
```

Comment:
Are these always uncapitalized? Can we go case `i`nsentitive for hex?
```suggestion
const entityIdRegex = /^([a-f]|[0-9]){8}-([a-f]|[0-9]){4}-([a-f]|[0-9]){4}-([a-f]|[0-9]){4}-([a-f]|[0-9]){12}$/i;
```

Reply:
We can poorly simplify this monster regexp, but I think it's worth it :)
```suggestion
const entityIdRegex = /^([a-f\d]){8}-([a-f\d]){4}-([a-f\d]){4}-([a-f\d]){4}-([a-f\d]){12}$/;
```

Reply:
We could go even further, but it might lack redability
```suggestion
const entityIdRegex = /^[a-f\d]{8}-([a-f\d]{4}-){3}[a-f\d]{12}$/i;
```

Reply:
> Are these always uncapitalized? Can we go case `i`nsentitive for hex?

From the Wikipedia entry on UUID:
```
RFC 4122 Section 3 requires that the characters be generated in lower case, while being case-insensitive on input.
```
So I think we can be forgiving on case.

Reply:
c90ae71

Good stuff @matibat !